lib/Agrammon/DataSource/CSV.pm6

```diff
@@ -44,6 +44,27 @@ class Agrammon::DataSource::CSV {
         }
     }
 
+    method from-csv($simulation-name, $dataset-id, $csv-data) {
+        my @group = $csv-data.split("\n");
+        my $input = Agrammon::Inputs.new(:$simulation-name, :$dataset-id);
+        for @group.map(*.split(';')) {
```

Comment:
It's maybe worth avoiding the temporary array here and just inlining the `$csv-data.split("\n")` in place of the group. (Also, it saves me complaining that `@group` is a slightly odd name for one record/line in the CSV file).

Reply:
:-) ... done.